src/main/java/com/google/sps/servlets/UpdateHabitServlet.java

```diff
@@ -27,7 +27,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
         Transaction transaction = datastore.newTransaction();
           Entity task = transaction.get(habitEntityKey);
-          if (task != null) {
+          System.out.println("Entity task = transaction.get(habitEntityKey) : " + task );
+          System.out.println(task.getBoolean("isComplete"));
+          if (task.getBoolean("isComplete")) {
+            transaction.put(Entity.newBuilder(task).set("isComplete", false).build());
```

Comment:
Double check if this is possible -- it's possible it isn't, but with most object classes, you can just say task.toBuilder() instead of Entity.newBuilder(task)

Reply:
the code doesnt seem to like it... it says "ToBuilder() is undefined for the type Entity"